### DISCUSSION TITLE
Fire Balls

### DISCUSSION THREAD

**Discussion by tsuza** [nosoop/SM-TFCustAttr#6] [Created attributes]
# Fireballs Attribute

This [custom attribute](https://github.com/Zabaniya001/att_fireballs) lets you throw fireballs! You've become a wizard, Harry!

You basically have a fuel system. Every 25% you can throw a fireball. The damage and speed depend on how much fuel you have. The more fuel you have, the more damage but also less speed it has. The damage fluctuates ( so like, it can deal from 300 dmg up to 500 dmg ), however, the more fuel it has the less it fluctuates.

Here is a [video](https://youtu.be/x2La9ud_KRw) if you're interested to see how it works first.

It was mainly made with Freak Fortress 2 in mind, so it might not fit TF2's normal gameplay.

## How to apply the attribute

Shove it inside tf_custom_attributes.txt if you want to replace a normal weapon's model ( or anything else that supports custom attributes ) or in the Custom Attributes section inside a custom weapon's cfg if you use Custom Weapons X.

```
  "fire wizard spells rage"	"max_rage=2000.0 max_damage_25%=500.0 max_damage_50%=1000.0 max_damage_75%=1500.0 max_damage_100%=2000.0"
```

These are its arguments. You can change 'em however it pleases you. 
```
- max_rage       <- The total amount of damage that the player has to deal to arrive to 100%.
- max_damage_25  <- The maximum amount of dmg it can deal at 25%.  | Speed = Rocket Speed       | Fluctuation = 60% of the max damage
- max_damage_50  <- The maximum amount of dmg it can deal at 50%.  | Speed = Rocket Speed * 0.8 | Fluctuation = 40 of the max damage
- max_damage_75  <- The maximum amount of dmg it can deal at 75%.  | Speed = Rocket Speed * 0.7 | Fluctuation = 20% of the max damage
- max_damage_100 <- The maximum amount of dmg it can deal at 100%. | Speed = Rocket Speed * 0.6 | Fluctuation = 10% of the max damage
```

## Technical Details

If you've taken a look at the source code you might be wondering why I store everything rather than getting them just whenever I need them. Well, for optimization reasons. I want to avoid any lags since OnTakeDamageAlive & the rest might get called a lot, especially when it's 32/32 players. Sure, it might cause inconsistencies in certain if something happens, but you gotta do what you gotta do.

## Dependencies

[TF2 Wearables](https://github.com/nosoop/sourcemod-tf2wearables)

[Stocksoup](https://github.com/nosoop/stocksoup) if you want to compile it yourself.

#

Let me know if there are any issues.

**Comment by nosoop** [comment]
Looks like you forgot download links for this and #7.

**Reply by tsuza** [reply]
Ah goddammit. Thank you for making me notice.
